test/unit/test_morphology.cpp

```diff
@@ -19,21 +17,38 @@
 
 #include "algorithms.hpp"
 #include "io/sepval.hpp"
+#include "morph/mbranch.hpp"
+#include "morph/em_morphology.hpp"
 #include "util/range.hpp"
 #include "util/span.hpp"
 #include "util/strprintf.hpp"
 
-// Forward declare non-public functions that are used internally to build
-// morphologies so that they can be tested.
-namespace arb { namespace impl {
-    std::vector<mbranch> branches_from_parent_index(const std::vector<arb::msize_t>&, const std::vector<point_prop>&, bool);
-}}
-
 template <typename T>
 std::ostream& operator<<(std::ostream& o, const std::vector<T>& v) {
     return o << "[" << arb::io::csv(v) << "]";
 }
 
+// Test basic functions on properties of mpoint
+TEST(morphology, mpoint) {
+    using mp = arb::mpoint;
+
+    EXPECT_EQ(arb::distance(mp{0,0,0},mp{0   , 0   , 0   }), 0.);
+    EXPECT_EQ(arb::distance(mp{0,0,0},mp{3.14, 0   , 0   }), 3.14);
+    EXPECT_EQ(arb::distance(mp{0,0,0},mp{0   , 3.14, 0   }), 3.14);
+    EXPECT_EQ(arb::distance(mp{0,0,0},mp{0   , 0   , 3.14}), 3.14);
+    EXPECT_EQ(arb::distance(mp{0,0,0},mp{1   , 2   , 3   }), std::sqrt(14.));
```

Comment:
This clearly works, but using a Pythagorean quadruple here would make the test less contingent on choice of fp precision used in the implementation,  or other floating point weirdosity.  e.g., `mp{7, 14, 22}` vs. `27`.